ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadPartitions.java

```diff
@@ -173,21 +174,22 @@ private ReplicationState initialReplicationState() throws SemanticException {
   private TaskTracker forNewTable() throws Exception {
     Iterator<AddPartitionDesc> iterator = event.partitionDescriptions(tableDesc).iterator();
     while (iterator.hasNext() && tracker.canAddMoreTasks()) {
-      AddPartitionDesc addPartitionDesc = iterator.next();
-      tracker.addTask(addSinglePartition(table, addPartitionDesc));
-      if (iterator.hasNext() && !tracker.canAddMoreTasks()) {
-        ReplicationState currentReplicationState =
-                new ReplicationState(new PartitionState(table.getTableName(), addPartitionDesc));
-        updateReplicationState(currentReplicationState);
-      }
+      addPartition(iterator.next());
     }
     return tracker;
   }
 
+  private void addPartition(AddPartitionDesc addPartitionDesc) throws Exception {
+    tracker.addTask(tasksForAddPartition(table, addPartitionDesc));
+    ReplicationState currentReplicationState =
+        new ReplicationState(new PartitionState(table.getTableName(), addPartitionDesc));
+    updateReplicationState(currentReplicationState);
```

Comment:
Need to set the replication state only if max number of tasks reached and still have more partitions to load. Or else, need to set it to null if all partitions are added within same DAG. This change is needed for REPL progress log feature.

Reply:
done